### PR TITLE
Fix shellcode customization

### PR DESCRIPTION
**Pull request by frederictobiasc**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Currently, shellcode customization is not implemented for all architectures. This pull request implements this functionality for x86-64. I will add the test and further architectures in another pull request.

**Test plan**

**Closing issues**

closes https://github.com/radareorg/radare2/issues/14648

### CODE REVIEW

**Comment by radare** [comment]
please address comments, rebase and add test

File: libr/egg/p/egg_exec.c [hunk 1]
```diff
@@ -96,6 +96,26 @@ static RBuffer *build (REgg *egg) {
 				break;
 			case 64:
 				sc = x86_64_linux_binsh;
+				if (shell && *shell) {
+					int len = strlen(shell);
```
Comment (radare) [inline libr/egg/p/egg_exec.c:100]: add space before every ( and around = , whats left to not be a draft?

**Comment by radare** [comment]
Ping

**Comment by radare** [comment]
can you fix the indentation issues? i think we're good to merge this before adding support for the other archs

**Comment by radare** [comment]
otherwise ill move to 4.5 unless you ping back

**Comment by frederictobiasc** [comment]
Hey, please excuse my late reaction. I was offline in the past week. I'll open a new pull request for the remaining architectures and the test.

**Comment by radare** [comment]
cool! thanks! and welcome back

**Comment by frederictobiasc** [comment]
You may merge now. I'm currently preparing a proposal to refactor the shellcode compilation.

**Pull request merged**